src/components/FileUpload/FileUpload.tsx

```diff
@@ -44,7 +44,8 @@ export const FileUpload: React.FC<FileUploadProps> = ({
 	passwordIncorrectText = 'Invalid password. Please try again.',
 	passwordConfirmButtonText = 'Confirm',
 	passwordCancelButtonText = 'Cancel',
-	passwordPlaceholderText = 'Password'
+	passwordPlaceholderText = 'Password',
+	invalidFileFormatErrorMessage = 'Invalid file format. Please try a different one.'
```

Comment:
The message should be something like `Unsupported file format. Supported formats:`. And then whenever this message is used inside the function you need to construct a correct message by appending the actual supported formats.

On line 34, we should reassign `acceptedFilePattern: _acceptedFilePattern`.

You need to move the `acceptedFilePattern ?? ...` (lines 145-148), assign it to a constant somewhere up in the function so we can use it.

```const acceptedFilePattern = _acceptedFilePattern ?? .. ```

And then extract the supported extensions, something like `Object.values(acceptedFilePattern).flat().join(', ')` and use it in the error message.

So the resulting error message will be for example `Unsupported file format. Supported formats: .png, .jpg, .jpeg, .gif, .pdf`

Reply:
Originally I had it like that, but than I thought If it wasn't more convenient if you could completely customise the error message at the time you implement/use this component. Especially for the sake of the translations. Because if the supported formats will be added automatically to the error message, you need to know how the translated error message must look when you pass it to the component via props so it wouldn't at the end look like something like this: _We are sorry but we cannot except this file's format. .png, .jpg, .jpeg, .gif, .pdf_. So you have to know, that the the invalid formats will be appended to the translated error message, so the error message must end with something like _Supported formats:_

But I couldn't decide for quite a long time which solution is better :D. I'll change it the way you suggest.

Reply:
I gave up on the idea that phoenix components would be some kind of open source project usable also for projects outside of the company a long time ago 😃 And since it's only for our purposes, the idea behind prop `unsupportedFileFormatErrorMessage` is not so that you would put there some, any sentence. It's there only for translations 😊 So the sentence will be like this every time. I don't see a reason why would we customise it any further.